gapic-common/lib/gapic/nonstandard_lro/operation.rb

```diff
@@ -0,0 +1,294 @@
+      #
+      # @return [Object, nil] The result of the operation or an error.
+      def results
+        return error if error?
+        return response if response?
+      end
+
+      ##
+      # Returns the name of the operation, if specified.
+      #
+      # @return [String, nil] The name of the operation.
+      #
+      def name
+        @operation.send @operation_name_field if @operation.respond_to? @operation_name_field
+      end
+
+      ##
+      # Checks if the operation is done. This does not send a new api call, but checks the result of the previous api
+      # call to see if done.
+      #
+      # @return [Boolean] Whether the operation is done.
+      #
+      def done?
+        return status if !status.nil? == status
+
+        status == :DONE
+      end
+
+      ##
+      # Checks if the operation is done and the result is not an error. If the operation is not finished then this will
+      # return false.
+      #
+      # @return [Boolean] Whether a response has been returned.
+      #
+      def response?
+        done? && !error?
+      end
+
+      ##
+      # If the operation is completed successfully, returns the underlying operation object, otherwise returns nil.
+      #
+      # @return [Object, nil] The response of the operation.
+      def response
+        @operation if response?
+      end
+
+      ##
+      # Checks if the operation is done and the result is an error. If the operation is not finished then this will
+      # return false.
+      #
+      # @return [Boolean] Whether an error has been returned.
+      #
+      def error?
+        done? && (!err.nil? || !error_code.nil?)
+      end
+
+      ##
+      # If the operation response is an error, the error will be returned, otherwise returns nil.
+      #
+      # @return [Object, nil] The error object.
+      #
+      def error
+        return unless error?
+        err || OpenStruct.new(code: error_code, message: error_msg)
+      end
+
+      ##
+      # Reloads the operation object.
+      #
+      # @param options [Gapic::CallOptions, Hash] The options for making the RPC call. A Hash can be provided
+      # to customize the options object, using keys that match the arguments for {Gapic::CallOptions.new}.
+      #
+      # @return [Gapic::NonstandardLro::Operation] Since this method changes internal state, it returns itself.
+      #
+      def reload! options: nil
+        return self if done?
+
+        @on_reload_callbacks.each { |proc| proc.call self }
+
+        request_hash = @request_values.transform_keys(&:to_sym)
+        @operation_copy_fields.each do |field_from, field_to|
+          request_hash[field_to.to_sym] = @operation.send field_from.to_s if @operation.respond_to? field_from.to_s
+        end
+
+        options = merge_options options, @options
+
+        ops = @client.send @polling_method_name, request_hash, options
+        ops = ops.operation if ops.is_a? Gapic::Rest::BaseOperation
+
+        @operation = ops
```

Comment:
Can we do something to break the name coupling on this instance variable from the base class? (Maybe create a protected setter in the base class.)

Reply:
I've added a protected setter. Protected `attr_writer` did not work for me however.

Reply:
Added a protected attr_writer.